tests/test_mathematical.py

```diff
@@ -2147,3 +2148,237 @@ def test_inplace_remainder(dtype):
     dp_a %= 4
 
     assert_allclose(dp_a, np_a)
+
+
+@pytest.mark.parametrize(
+    "dtype", get_all_dtypes(no_bool=True, no_none=True, no_complex=True)
+)
+def test_inplace_floor_divide(dtype):
+    size = 21
+    np_a = numpy.arange(size, dtype=dtype)
+    dp_a = dpnp.arange(size, dtype=dtype)
+
+    np_a //= 4
+    dp_a //= 4
+
+    assert_allclose(dp_a, np_a)
+
+
+@pytest.mark.parametrize(
+    "order_pair", [("C", "C"), ("C", "F"), ("F", "C"), ("F", "F")]
+)
+@pytest.mark.parametrize(
+    "shape_pair",
+    [
+        ((4,), (4,)),
+        ((4,), (4, 2)),
+        ((2, 4), (4,)),
+        ((2, 4), (4, 3)),
+        ((1, 2, 3), (1, 3, 5)),
+        ((4, 2, 3), (4, 3, 5)),
+        ((1, 2, 3), (4, 3, 5)),
+        ((2, 3), (4, 3, 5)),
+        ((4, 2, 3), (1, 3, 5)),
+        ((4, 2, 3), (3, 5)),
+        ((1, 1, 4, 3), (1, 1, 3, 5)),
+        ((6, 7, 4, 3), (6, 7, 3, 5)),
+        ((6, 7, 4, 3), (1, 1, 3, 5)),
+        ((6, 7, 4, 3), (1, 3, 5)),
+        ((6, 7, 4, 3), (3, 5)),
+        ((6, 7, 4, 3), (1, 7, 3, 5)),
+        ((6, 7, 4, 3), (7, 3, 5)),
+        ((6, 7, 4, 3), (6, 1, 3, 5)),
+        ((1, 1, 4, 3), (6, 7, 3, 5)),
+        ((1, 4, 3), (6, 7, 3, 5)),
+        ((4, 3), (6, 7, 3, 5)),
+        ((6, 1, 4, 3), (6, 7, 3, 5)),
+        ((1, 7, 4, 3), (6, 7, 3, 5)),
+        ((7, 4, 3), (6, 7, 3, 5)),
+        ((1, 5, 3, 2), (6, 5, 2, 4)),
+        ((5, 3, 2), (6, 5, 2, 4)),
+        ((1, 3, 3), (10, 1, 3, 1)),
+    ],
+)
+def test_matmul(order_pair, shape_pair):
```

Comment:
Please create `TestMatmul` class and incorporate `test_matmul` , `test_matmul_dtype`,  `test_matmul_order` ,`test_matmul_strided`  and `test_matmul_out` functions into it